src/main/java/com/myhome/controllers/HouseController.java

```diff
@@ -79,14 +88,13 @@
   )
   public ResponseEntity<GetHouseDetailsResponse> getHouseDetails(@PathVariable String houseId) {
     log.trace("Received request to get details of a house with id[{}]", houseId);
-    if (!houseService.getHouseDetailsById(houseId).isPresent()) {
+
+    Optional<CommunityHouse> houseDetail = Optional.of(houseService.getHouseDetailsById(houseId)).orElse(null);
```

Comment:
This is incorrect.
Please look on other controllers how to handle optional flow.